unime/src/routes/badges/[id]/+page.svelte

```diff
@@ -64,6 +64,32 @@
     return JSON.stringify(object, null, 2);
   };
 
+  // TODO: This is a HORRIBLE solution to determine the connection_id by the non-unique "issuer name".
+  // It is a TEMPORARY solution and should only be used in DEMO environments,
+  // since we currently lack a unique identitfier to distinguish connections.
+  function determineConnectionId(): string | null {
+    // First collect possible sources of the issuer name
+    const name_from_credential: string | undefined = credential.data.issuer?.name;
+    const name_from_oid4vc = credential.issuer_name;
+    // We prefer the name from the credential, but fallback to the issuer name during oid4vc process
+    const issuer_name = name_from_credential ?? name_from_oid4vc;
+
+    if (issuer_name) {
+      // base64url encode
+      const connectionId = btoa(issuer_name).replace('+', '-').replace('/', '_');
+      // verify that the connection exists
+      if ($state.connections.find((c) => c.id === connectionId)) {
```

Comment:
Maybe you can use the some function here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some